src/js/hooks/cycleCount/useCountStep.jsx

```diff
@@ -575,15 +576,77 @@ const useCountStep = () => {
     }
   };
 
+  const createCustomItemsFromImport = (items) => (items
+    ? items.map((item) => ({
+      ...item,
+      countIndex: 0,
+      id: _.uniqueId('newRow'),
+      custom: true,
+      inventoryItem: {
+        lotNumber: item.lotNumber,
+        expirationDate: item.expirationDate,
+      },
+      product: {
+        id: item.product.id,
+        productCode: item.product.productCode,
+      },
+    }))
+    : []);
+
+  const removeItemFromCycleCounts = (cycleCounts, cycleCountId, itemId) => ({
+    ...cycleCounts,
+    [cycleCountId]: cycleCounts[cycleCountId]
+      .filter((item) => item.cycleCountItemId !== itemId),
+  });
+
+  const mergeImportItems = (originalItem, importedItem) => ({
+    ...originalItem,
+    ...importedItem,
+    inventoryItem: {
+      product: {
+        id: importedItem?.product?.id || originalItem?.inventoryItem?.product?.id,
+        name: importedItem?.product?.name || originalItem?.inventoryItem?.product?.name,
+        productCode: importedItem?.product?.productCode
+            || originalItem?.inventoryItem?.product?.productCode,
+      },
+      lotNumber: importedItem?.lotNumber || originalItem?.inventoryItem?.lotNumber,
+      expirationDate: importedItem?.expirationDate || originalItem?.inventoryItem?.expirationDate,
+    },
+  });
+
   const importItems = async () => {
     try {
       show();
       const response = await cycleCountApi.importCycleCountItems(importFile, currentLocation?.id);
-      console.log(response);
+      let cycleCounts = _.groupBy(response.data.data, 'cycleCountId');
+      tableData.current = tableData.current.map((cycleCount) => ({
+        ...cycleCount,
+        cycleCountItems: [
+          ...cycleCount.cycleCountItems
+            .map((item) => {
+              const correspondingImportItem = cycleCounts[cycleCount.id]?.find(
+                (cycleCountItem) => cycleCountItem.cycleCountItemId === item.id,
+              );
+
+              if (correspondingImportItem) {
+                // Remove items from the import that have a corresponding item
+                // in the current cycle count. It allows us to treat items with
+                // the wrong ID as new rows that do not already exist.
```

Comment:
could you explain with an example what's the purpose of that? If a corresponding item exists, we should update its values instead of removing it, but maybe I don't understand the point of those lines?